### PR TITLE
Security api umbrella

### PR DESCRIPTION
**Pull request by chicco785**

### CODE REVIEW

**Comment by chicco785** [comment]
this should be ready for review, i am ccing the external people interested.

@anmunoz
@taliaga 
@omerozdemiratos

File: recipes/security/api-umbrella/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+version: '3.3'
+
+services:
+
+  api-umbrella:
+    image: martel/api-umbrella:${API_UMBRELLA_VERSION:-0.14.4-1-fiware}
```
Comment (taliaga) [inline recipes/security/api-umbrella/docker-compose.yml:6]: what a tag! :p

**Comment by omerozdemiratos** [comment]
Thanks Federico, I´ll check the details.

File: recipes/security/readme.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Security Management
+
+Within FIWARE security is by different services that provide the following
```
Comment (taliaga) [inline recipes/security/readme.md]: missing verb after security

File: recipes/security/readme.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+All the above elements combined together provide you with an AAA solution for
+FIWARE APIs.
+
+In the recipes, at the time being, we cover only the ongoing implementation
+of the PEP Proxy based on API Umbrella. This is due to the fact that:
+
+* KeyRock is undergoing strong developments and new release will be announced
+    soon, and the current FIWARE Lab IDM can be used for any project without
+    need to deploy your own instance.
+* PDP is required only in complex scenarios, and the PDP available n FIWARE Lab
```
Comment (taliaga) [inline recipes/security/readme.md]: typo: in

**Comment by chicco785** [comment]
hi omer,
if you read the docs, you will see you have also examples related to how to use the API to register backends APIs and publishing them.

still, there is a part of manual process related to get at least the ADMIN and API KEY.

File: recipes/security/api-umbrella/readme.md [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+In case you haven't done it yet for other recipes, deploy `backend` and
+`frontend` networks as described in the
+[installation guide](../../installation.md#creating-the-networks).
+
+API Umbrella needs a mongo database for its backend. If you have already
+deployed Mongo within your cluster and would like to reuse that database, you
+can skip the next step (deploying backend).
+You will just need to pay attention to the variables
+you define for Orion to link to Mongo, namely, `MONGO_SERVICE_URI`
+and `REPLICASET_NAME`. Make sure you have the correct values in `frontend.env`.
```
Comment (taliaga) [inline recipes/security/api-umbrella/readme.md]: this should be backend.env

File: recipes/security/api-umbrella/readme.md [hunk 5]
```diff
@@ -0,0 +1,469 @@
...
+```
+    $ sh deploy_back.sh
+
+    Creating config mongo-rs_mongo-healthcheck
+    Creating service mongo-rs_mongo
+    Creating service mongo-rs_controller
+```
+
+Beside that, given that Ruby driver for MongoDB is not supporting service
+discovery, you will need to have expose the ports of the MongoDB server on the
```
Comment (taliaga) [inline recipes/security/api-umbrella/readme.md]: minor typo: remove have before expose

File: recipes/security/api-umbrella/readme.md [hunk 6]
```diff
@@ -0,0 +1,469 @@
...
+
+In the menu select "Users->Api Users" click on the username
+`web.admin.ajax@internal.apiumbrella` and copy the Api
+Key (of course you can create new ones instead of reusing API Umbrella
+defaults).
+
+Create a simple API to test that everything works:
+
+```
+$ curl -k -X POST "https://<your-cluster-manager-ip>/api-umbrella/v1/apis" -H "X-Api-Key: <your-API-KEY>" -H "X-Admin-Auth-Token: <your-admin-auth-token>" -H "Accept: application/json" -H "Content-Type: application/json" -d @- <<EOF
```
Comment (taliaga) [inline recipes/security/api-umbrella/readme.md]: I'd suggest placing them in files next to readme.

File: recipes/security/api-umbrella/readme.md [hunk 7]
```diff
@@ -0,0 +1,469 @@
...
+Now, let's create the admin user in API Umbrella. As first thing,
+get the IP of your master node:
+
+```
+    $ docker-machine ip ms-manager0
+```
+
+Open the browser at the following endpoint: [http://ip/admin](http://ip/admin).
+
+Unless you also created certificates for your server, API Umbrella,
```
Comment (taliaga) [inline recipes/security/api-umbrella/readme.md]: At some point around here I'd add pointers to API umbrella's doc

File: recipes/security/api-umbrella/readme.md [hunk 8]
```diff
@@ -0,0 +1,469 @@
...
+However, to access the container from outside of the overlay network (for
+example from the host) you would need to access the ip of the container's
+interface to the `docker_gwbridge`. It seem there's no easy way to get that
+information from the outside (see
+[this open issue](https://github.com/docker/libnetwork/issues/1082).
+In the walkthrough, we queried API Umbrella through one of the swarm nodes
+because we rely on docker ingress network routing the traffic all the way
+to one of the containerized API Umbrella services.
+
+## Open interesting issues
```
Comment (taliaga) [inline recipes/security/api-umbrella/readme.md:468]: Shall we keep?

**Review by Cerfoglg — APPROVED** [review]

File: recipes/security/api-umbrella/readme.md (hunk 6 above)
Comment (chicco785) [reply recipes/security/api-umbrella/readme.md]: not sure what you mean

File: recipes/security/api-umbrella/readme.md (hunk 4 above)
Comment (chicco785) [reply recipes/security/api-umbrella/readme.md]: it's correct frontend. if you already deployed backend, you need only to set values in frontend.

File: recipes/security/api-umbrella/readme.md (hunk 8 above)
Comment (chicco785) [reply recipes/security/api-umbrella/readme.md:468]: probably not

**Pull request merged**